### PR TITLE
[INF-227] Enforce correct response value complies with base type

### PR DESCRIPTION
**Pull request by KirylHatalski**
Ticket: <a class="link" href='https://oat-sa.atlassian.net/browse/INF-227'>INF-227</a>

## What's Changed

- Fixed test creator ability to set incorrect type of data on correct answer step

## Demo
image

## How to test
- Create a test with an item with a Text Entry interaction with Response base type set as integer.
- Try to set correct answer with float or string
- Value should be converted to setted type

[Kitchen](http://test-kiril.playground.kitchen.it.taocloud.org:40351/tao/Main/login)

### CODE REVIEW

File: views/js/qtiCreator/helper/textEntryConverterHelper.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+], function ($, locale, converter) {
+    'use strict';
+
+   function textEntryConverterHelper(value, attributes) {
+       const numericBase = attributes.base || 10;
+       const convertedValue = converter.convert(value.trim());
+       switch (attributes.baseType) {
+           case 'integer':
+               value = locale.parseInt(convertedValue, numericBase);
+               return isNaN(value) ? '' : value;
```
Comment (andreluizmachado) [inline views/js/qtiCreator/helper/textEntryConverterHelper.js:32]:
```suggestion
               return isNaN(value) ? 0 : value;
```

File: views/js/qtiCreator/widgets/interactions/textEntryInteraction/states/Correct.js [hunk 2]
```diff
@@ -32,8 +33,10 @@ define([
         const correctResponse = stringResponseHelper.getCorrectResponse(response);
 
         $container.find('tr[data-edit=correct] input[name=correct]').focus().val(correctResponse);
-        $container.on('keyup.correct', 'tr[data-edit=correct] input[name=correct]', function () {
-            const value = $(this).val();
+        $container.on('blur.correct', 'tr[data-edit=correct] input[name=correct]', function () {
+            const $input = $(this);
+            let value = textEntryConverterHelper($input.val(), response.attributes);
```
Comment (viktar-dzmitryieu-tao) [inline views/js/qtiCreator/widgets/interactions/textEntryInteraction/states/Correct.js:38]: could be `const` instead

**Review by viktar-dzmitryieu-tao — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [X] New code is respecting code style rules
- [X] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [X] Feature is working correctly on my local machine (if applicable)
- [X] Acceptance criteria are respected
- [X] Pull request title and description are meaningful
- [X] Pull request's target is not `master`

File: views/js/qtiCreator/helper/textEntryConverterHelper.js (hunk 1 above)
Comment (KirylHatalski) [reply views/js/qtiCreator/helper/textEntryConverterHelper.js:32]: Had several tries to playaround with that and seems like helper should keep all unparsed values as empty string.
All force conversion to this exact types like you suggested will give a problem with exceptional cases, cuz 0 and .0 is valid numbers for current logic.

File: views/js/qtiCreator/widgets/interactions/textEntryInteraction/states/Correct.js (hunk 2 above)
Comment (KirylHatalski) [reply views/js/qtiCreator/widgets/interactions/textEntryInteraction/states/Correct.js]: Fixed, thanks

**Review by andreluizmachado — COMMENTED** [review]
Before my approval, can you check why [ci](https://github.com/oat-sa/extension-tao-itemqti/actions/runs/7529882604/job/20495084989?pr=2433) is failing?

**Review by andreluizmachado — APPROVED** [review]

**Review by viktar-dzmitryieu-tao — APPROVED** [review]
LGTM!

**Review by viktar-dzmitryieu-tao — APPROVED** [review]

**Pull request merged**